### PR TITLE
Remove unnecessary boxing / unboxing (Java 5 type of warnings)

### PR DESCRIPTION
**Pull request by sigee**

### CODE REVIEW

**Comment by sigee** [comment]
@psiinon,
I have added the comments to almost every class under org.parosproxy.paros packages package except:
- org.parosproxy.paros.extension.encoder.Base64
- org.parosproxy.paros.extension.history.HistoryFilter
- org.parosproxy.paros.extension.option.DatabaseParam
- org.parosproxy.paros.extension.option.ProxyExcludedDomainsTableModel

because these are ZAP and its related class files or "This code is from Public Domain" and any ot them had ZAP: YYY/MM/DD type of comments however they have been changed.
Should I add the comments to these files also or it is not needed?

**Comment by thc202** [comment]
It's not necessary, only the Paros classes need the ZAP comment.

File: src/org/apache/commons/httpclient/HttpMethodDirector.java [hunk 1]
```diff
@@ -410,7 +410,7 @@ private void applyConnectionParams(final HttpMethod method) throws IOException {
             param = this.conn.getParams().getParameter(HttpConnectionParams.SO_TIMEOUT);
         }
         if (param != null) {
-            timeout = ((Integer)param).intValue();
+            timeout = (Integer) param;
```
Comment (thc202) [inline src/org/apache/commons/httpclient/HttpMethodDirector.java]: Would you mind reverting this change? (Be good to keep the changes to these forked classes to a minimum necessary for our use cases.)

Comment (sigee) [reply src/org/apache/commons/httpclient/HttpMethodDirector.java]: @thc202,
You are right. I'll revert this one.

**Comment by sigee** [comment]
@thc202,
I've added "If it is possible avoid introduce the un/boxings in new code." to the Dev guidelines wiki page.

File: src/org/apache/commons/httpclient/HttpMethodDirector.java [hunk 2]
```diff
@@ -410,7 +410,7 @@ private void applyConnectionParams(final HttpMethod method) throws IOException {
             param = this.conn.getParams().getParameter(HttpConnectionParams.SO_TIMEOUT);
         }
         if (param != null) {
-            timeout = ((Integer)param).intValue();
+            timeout = ((Integer) param).intValue();
```
Comment (thc202) [inline src/org/apache/commons/httpclient/HttpMethodDirector.java]: Could this change be reverted as well?

**Comment by thc202** [comment]
Thanks for adding the guideline entry.

**Comment by thc202** [comment]
Thanks!

**Pull request merged**